src/pivot-table/components/grids/LeftGrid.tsx

```diff
@@ -71,11 +71,9 @@ const LeftGrid = ({
     grid: { divider },
   } = useStyleContext();
 
-  useOnPropsChange(() => {
-    if (leftGridRef.current) {
-      leftGridRef.current.forEach((list) => list?.resetAfterIndex(0, false));
-    }
-  }, [dataModel, width, height, leftDimensionData, leftGridRef, contentCellHeight]);
+  useResetListCache(leftGridRef, dataModel, leftDimensionData);
+
+  useResetListCacheAndRerender(leftGridRef, width, height, contentCellHeight);
```

Comment:
When the width/height changes, the internal cache of react-windows is stale. This resets that cache and forced the grid to rerender.

It was actually already done in the `DataGrid` which could cause the "fix" in this PR put the `TopGrid` and `DataGrid` out of sync. As `TopGrid` would render with stale width values.